src/bitwarden_workflow_linter/rules/check_pr_target.py

```diff
@@ -0,0 +1,80 @@
+"""A Rule to enforce check-run is run when workflow uses pull_request_target."""
+
+from typing import Optional, Tuple
+
+from ..models.workflow import Workflow
+from ..rule import Rule
+from ..utils import LintLevels, Settings
+
+
+class RuleCheckPrTarget(Rule):
+    def __init__(self, settings: Optional[Settings] = None) -> None:
+        """
+        To ensure pull_request_target is safe to use, the check-run step is added 
+        to all jobs as a dependency.
+
+        Once a branch is pushed to Github, it already opens up a vulnerability 
+        even if the check-run scan fails to detect this. 
+        
+        In order to prevent a vulnerable branch from being used for an attack 
+        prior to being caught through vetting, all pull_request_target workflows 
+        should only be run by users with appropriate permissions.  
+
+        This greatly reduces the risk as community contributors can't use a fork to run a compromised workflow that uses pull_request_target.
+        """
+        self.message = "A check-run job must be included as a direct job dependency when pull_request_target is used and the workflow may only apply to runs on the main branch"
+        self.on_fail = LintLevels.WARNING
+        self.compatibility = [Workflow]
+        self.settings = settings
+    
+    def targets_main_branch(self, obj:Workflow) -> Tuple[bool]:
+        branches_list = [
+            branch
+            for key, value in obj.on.items() if value
+            for key2, value2 in value.items() if key2 == 'branches'
+            for branch in value2
+            ]
+        if any(branch != 'main' for branch in branches_list):
+            return False
+        else:
+            return True
```

Comment:
Does this check the branch of the trigger?

Reply:
yes it does, I switched it to check the branch it's targeting